CONTRIBUTING.md

```diff
@@ -0,0 +1,145 @@
+- Choose a short, meaningful title
+- Start the issue description with a quick summary of the problem you encountered
+- Include steps to reproduce the issue
+- State how you expected the program to behave, and how it actually behaved
+- If applicable, write down stuff you tried to solve the problem that didn't work
+- Feel free to include personal thoughts about why you think this issue is occurring
+
+### Proposing features
+
+As for bug reports, we use GitHub issues to track feature requests. So, if you want to propose a feature or enhancement, just [open a new issue](https://github.com/riggraz/astuto/issues/new). As always, check whether that feature has already been proposed before posting.
+
+### Coding
+
+In order to contribute code, you should use the usual GitHub fork workflow. See [this section](#workflow) if you aren't comfortable with it yet.
+
+The following is a list of some guidelines to take into consideration when contributing code:
+
+- If you're going to fix a bug or implement a feature that has already been documented with an issue, please let the community know by posting a comment on that issue. In this way, we avoid the (unfortunate) possibility of two people working on the same thing at the same time.
+- If you want to fix a bug or implement a new feature that hasn't already been documented with an issue, please [open an issue](https://github.com/riggraz/astuto/issues/new) before.
+- Follow styleguides for both [commit messages](#git-commit-messages) and [code](#coding-conventions)
+- When you're done, open a pull request and explain clearly what it is about. We also encourage you to link to the issue it fixes.
+
+### Getting involved in the community
+
+We'd love to hear your opinion both on small feature additions and on long-term strategic discussions. Indeed, you can contribute to Astuto just by dropping a comment to let us know your point of view.
+
+In particular, some [issues](https://github.com/riggraz/astuto/issues?q=is%3Aissue+is%3Aopen+label%3Adiscussion) and [pull requests](https://github.com/riggraz/astuto/pulls?q=is%3Aopen+is%3Apr+label%3Adiscussion) are labeled "discussion". That label signifies that further discussion is required to move forward the development process: thus, we encourage you to express your opinion especially on these topics.
+
+## Styleguides
+
+### Git commit messages
+
+```
+Short summary of the commit
+
+More throughout explaination of what the commit does and the reasoning behind it (optional)
+```
+
+For commit messages, please follow these broad guidelines:
+
+- Separate subject from body with a blank line
+- The subject must use present tense ("Add feature" not "Added feature") and imperative mood ("Fix bug" not "Fixes bug")
+- Limit the subject to about 72 characters
+- Try to be both coincise and clear
+
+### Coding conventions
+
+#### General coding conventions
+
+- Write clean code
```

Comment:
- If you add or improve a feature, please add associated spec
- Run specs locally before pushing

Reply:
I'd add that we *strongly encourage* to add the associated spec, but I wouldn't make it compulsory as it may discourage one to contribute. And, moreover, a lot of code (I'm thinking about JS in particular) as of now is not covered by tests so it may be non coherent to say that we require tests.

Reply:
Good enough :) It's just good to tell it as a little reminder (if people start to push un-covered code, it means somebody have to check it absolutely by hand, or face consequences ^^')